pkg/resources/grant_privileges_to_share.go

```diff
@@ -0,0 +1,474 @@
+		},
+	}
+}
+
+func ImportGrantPrivilegesToShare() func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		id, err := ParseGrantPrivilegesToShareId(d.Id())
+		if err != nil {
+			return nil, err
+		}
+		if err := d.Set("share_name", id.ShareName.Name()); err != nil {
+			return nil, err
+		}
+		if err := d.Set("privileges", id.Privileges); err != nil {
+			return nil, err
+		}
+
+		switch id.Kind {
+		case OnDatabaseShareGrantKind:
+			if err := d.Set("database_name", id.Identifier.Name()); err != nil {
+				return nil, err
+			}
+		case OnSchemaShareGrantKind:
+			if err := d.Set("schema_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		// case OnFunctionShareGrantKind:
+		//	if err := d.Set("function_name", id.Identifier.FullyQualifiedName()); err != nil {
+		//		return nil, err
+		//	}
+		case OnTableShareGrantKind:
+			if err := d.Set("table_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnAllTablesInSchemaShareGrantKind:
+			if err := d.Set("all_tables_in_schema", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnTagShareGrantKind:
+			if err := d.Set("tag_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		case OnViewShareGrantKind:
+			if err := d.Set("view_name", id.Identifier.FullyQualifiedName()); err != nil {
+				return nil, err
+			}
+		}
+
+		return []*schema.ResourceData{d}, nil
+	}
+}
+
+func CreateGrantPrivilegesToShare(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	id := createGrantPrivilegesToShareIdFromSchema(d)
+	log.Printf("[DEBUG] created identifier from schema: %s", id.String())
+
+	err := client.Grants.GrantPrivilegeToShare(ctx, getObjectPrivilegesFromSchema(d), getShareGrantOn(d), sdk.NewAccountObjectIdentifier(id.ShareName.Name()))
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "An error occurred when granting privileges to share",
+				Detail:   fmt.Sprintf("Id: %s\nShare name: %s\nError: %s", id.String(), id.ShareName, err.Error()),
+			},
+		}
+	}
+
+	d.SetId(id.String())
+
+	return ReadGrantPrivilegesToShare(ctx, d, meta)
+}
+
+func UpdateGrantPrivilegesToShare(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
+	id, err := ParseGrantPrivilegesToShareId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+
+	if d.HasChange("privileges") {
+		before, after := d.GetChange("privileges")
```

Comment:
not important at all, but typically this would be called "old" and "new"

Reply:
changed